Marlin/src/module/planner.cpp

```diff
@@ -2086,36 +2137,111 @@ bool Planner::_populate_block(block_t * const block, bool split_move,
     if (millimeters)
       block->millimeters = millimeters;
     else {
-      block->millimeters = SQRT(
+      // Distance for interpretation of feedrate in accordance with LinuxCNC
+      #if ENABLED(FOAMCUTTER_XYUV) && LINEAR_AXES == 5
+        // Return the largest distance move from either X/Y or I/J plane
+        block->millimeters = SQRT(
+        _MAX(sq(steps_dist_mm.x) + sq(steps_dist_mm.y), sq(steps_dist_mm.i) + sq(steps_dist_mm.j))
+      #elif LINEAR_AXES >= 4
+        float distance_sqr = (
+      #else
+        block->millimeters = SQRT(
+      #endif
```

Comment:
This whole block of code (including the portions below which I didn't highlight) is _incredibly_ difficult to read and understand.

Reply:
If everyone agrees, I could  add the float distance_sqr for all cases, even LINEAR_AXES 3. Then the algorithm would be the same for all cases: first calculate distance_sqr, finally block->millimeters = SQRT(distance_sqr)

Using the macro `TERN_` would makes code more compact but reduces readability even further. 

I will try to propose a solution which uses new macros: 
```
#if HAS_ROTATIONAL_AXES9
  #define ROTATIONAL_AXES 9
#elif HAS_ROTATIONAL_AXES8
  #define ROTATIONAL_AXES 8
...
#endif
#define SECONDARY_LINEAR_AXES LINEAR_AXES - ROTATIONAL_AXES

#define XYZ_GANG   GANG_N(XYZ,I,J,K,U,V,W)
#define ROTATIONAL_AXES_GANG  GANG_N(ROTATIONAL_AXES,I,J,K,U,V,W)
#define SECONDARY_LINEAR_AXES_GANG(I,J,K,U,V,W)  GANG_N(SECONDARY_LINEAR_AXES,I,J,K,U,V,W)

#define XYZ_CODE  CODE_N(XYZ,X,Y,Z)
#define ROTATIONAL_AXES_CODE  CODE_N(ROTATIONAL_AXES,I,J,K,U,V,W)
#define SECONDARY_LINEAR_AXES_CODE(I,J,K,U,V,W)  CODE_N(SECONDARY_LINEAR_AXES,I,J,K,U,V,W)
```

Reply:
done